custom_components/nordnet/config_flow.py

```diff
@@ -1,64 +1,120 @@
 from __future__ import annotations
 
+import copy
 import logging
-from collections.abc import Mapping
 from typing import Any
 
+import aiohttp
 import voluptuous as vol
-from homeassistant.config_entries import CONN_CLASS_CLOUD_POLL
-from homeassistant.core import callback
+from homeassistant.config_entries import (CONN_CLASS_CLOUD_POLL, ConfigEntry,
+                                          ConfigFlow, OptionsFlow)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.data_entry_flow import FlowResult
 from homeassistant.helpers import selector
-from homeassistant.helpers.schema_config_entry_flow import (
-    SchemaConfigFlowHandler, SchemaFlowFormStep, SchemaFlowMenuStep)
 
 from .const import (DEFAULT_ACCOUNT_ID, DEFAULT_SESSION_LIFETIME,
                     DEFAULT_TRADING_START_TIME, DEFAULT_TRADING_STOP_TIME,
                     DEFAULT_UPDATE_INTERVAL, DOMAIN, PLATFORM)
+from .coordinator import Coordinator
 
 _LOGGER = logging.getLogger(__name__)
 
 # See https://github.com/home-assistant/core/blob/dev/homeassistant/helpers/selector.py
-OPTIONS_SCHEMA = vol.Schema(
+DATA_SCHEMA = vol.Schema(
     {
         vol.Required("username"): selector.TextSelector(),
         vol.Required("password"): selector.TextSelector({'type': 'password'}),
-        vol.Required("account_id", default=DEFAULT_ACCOUNT_ID): selector.NumberSelector({'mode': 'box', 'step': 1}),
-        vol.Required("trading_start_time", default=DEFAULT_TRADING_START_TIME): selector.TimeSelector(),
-        vol.Required("trading_stop_time", default=DEFAULT_TRADING_STOP_TIME): selector.TimeSelector(),
-        vol.Required("update_interval", default=DEFAULT_UPDATE_INTERVAL): selector.DurationSelector(),
-        vol.Required("session_lifetime", default=DEFAULT_SESSION_LIFETIME): selector.DurationSelector(),
+        vol.Required("account_id", description={"suggested_value": DEFAULT_ACCOUNT_ID}): selector.NumberSelector({'mode': 'box', 'step': 1}),
+        vol.Required("trading_start_time", description={"suggested_value": DEFAULT_TRADING_START_TIME}): selector.TimeSelector(),
+        vol.Required("trading_stop_time", description={"suggested_value": DEFAULT_TRADING_STOP_TIME}): selector.TimeSelector(),
+        vol.Required("update_interval", description={"suggested_value": DEFAULT_UPDATE_INTERVAL}): selector.DurationSelector(),
+        vol.Required("session_lifetime", description={"suggested_value": DEFAULT_SESSION_LIFETIME}): selector.DurationSelector(),
     }
 )
 
-CONFIG_SCHEMA = vol.Schema(
-    {
 
-    }
-).extend(OPTIONS_SCHEMA.schema)
+async def validate_connection(hass: HomeAssistant, user_input: dict) -> dict:
+    """
+    Validate connection and credentials to Nordnet API when creating or
+    updating a configuration entry
+    """
+    try:
+        c = Coordinator(hass, Coordinator.map_config(user_input))
```

Comment:
Should not start the coordinator here for these tests.

Reply:
Is it starting it? isn't that happening in `async_setup_entry` after it being added - i thought this would just create a throw-away instance to test authentication and then it will get garbage collected away and never seen again

Reply:
Not entirely sure but best practice would anyhow not be to start a coordinator for such test but connect with `async_create_clientsession`